### PR TITLE
add PROXY_REPO_SHA removing overloaded meaning of PROXY_TAG

### LINKED ISSUE #4157: Clear up confusion around proxy tag
PROXY_TAG is being used as the tag for the docker image and also as the sha of the istio/proxy to use as base for building the docker proxy image in this repo. This dual is wrong and confusing,

### PR DESCRIPTION
**Pull request by rkpagadala**
PROXY_TAG now refers to the docker image tag build from file pilot/docker/Dockerfile.proxy or pilot/docker/Dockerfile.proxy_debug
PROXY_REPO_SHA refers to the sha of istio/proxy, and is used to download the envoy binary used to build the proxy docker image referred above.
fixes #4157

### CODE REVIEW

File: install/updateVersion.sh [hunk 1]
```diff
@@ -50,17 +49,22 @@ EOF
   exit 2
 }
 
-source "$SRC_VERSION_FILE" || error_exit "Could not source versions"
+# set the default values
```
Comment (rkpagadala) [inline install/updateVersion.sh]: This came from istio.VERSION, missing this caused the post-submit failures

**Comment by ldemailly** [comment]
did you check the build that failed before will not this time, and matt's point about the release process?

**Comment by ldemailly** [comment]
that new error about 
 services "istio-sidecar-injector" not found
is recent but unrelated

**Comment by rkpagadala** [comment]
@ldemailly the way I tested it is by making sure that the old files generated and the files after my change are identical. Missed eureka and consul fixes in  https://github.com/istio/istio/pull/4181 for your consideration.

 About Matt's point we don't need any changes in daily release for the release process. There are unrelated but needed changes that I asked Charles to make.

**Pull request merged**